### PR TITLE
CI-515: Make it possible for users to add emojis to the Shoutout answers on the Results page

### PR DESCRIPTION
**Pull request by SerhiiBorozenets**
[CI-515](https://cloverpop-internship.atlassian.net/browse/CI-515)

[CI-515]: https://cloverpop-internship.atlassian.net/browse/CI-515?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by SerhiiBorozenets** [comment]
Code climate issues are not relevant

File: app/javascript/components/Pages/ResultsPage/EmojiPicker.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+const removeEmoji = (emojiObject, emojisArr, setEmojisArr, setSelectedEmoji, current_user, setEmojiObject) => {
+  axios.delete(`/api/v1/emojis/${emojiObject.id}`)
+    .then(res => {
+      if (res.data.message === 'success') {
+        const updatedData = emojisArr.map(item => {
+          if (item.emoji_code === emojiObject.emoji_code) {
+            const newUsers = item.users.filter(user => user.id !== current_user.id)
+            if (item.count < 2) return {};
```
Comment (rogergraves) [inline app/javascript/components/Pages/ResultsPage/EmojiPicker.js:44]: There are quite a lot of nested conditionals here...

**Review by rogergraves — COMMENTED** [review]
Please also check the CodeClimate issues

File: app/javascript/components/Pages/ResultsPage/EmojiPicker.js [hunk 2]
```diff
@@ -37,26 +37,25 @@ const createEmoji = (emojiObject, emojisArr, setEmojisArr, setSelectedEmoji, set
...
+          const newUsers = item.users.filter(user => user.id !== current_user.id)
+          return Object.assign({}, item, {
+            emoji_code: item.count > 1 ? item.emoji_code : null,
+            emoji_name: item.emoji_name,
+            count: item.count - 1,
+            users: newUsers
+          });
+        }  else {
+          return item;
+        }
```
Comment (rogergraves) [inline app/javascript/components/Pages/ResultsPage/EmojiPicker.js:53]: Maybe a way to remove another nesting layer:
```javascript
if (item.emoji_code !== emojiObject.emoji_code) return item;
```

File: app/javascript/components/helpers/helpers.js [hunk 3]
```diff
@@ -99,34 +99,24 @@ export function convertUsersToString(users) {
...
 export function usersEmoji(users, current_user, emojiObject) {
   const modifiedUsers = users.map(user => ({
-    ...user,
-    first_name: user.id === current_user.id ? 'you' : user.first_name
+    ...user, display_name: user.id === current_user.id ? 'you' : user.first_name
   }));
-
-  if (modifiedUsers.length === 0) {
+  const userCount = modifiedUsers.length;
+  if (userCount === 0) {
```
Comment (rogergraves) [inline app/javascript/components/helpers/helpers.js:105]: Maybe can remove the `{ }`?

**Review by rogergraves — APPROVED** [review]

**Review by Maryna-Harasko — APPROVED** [review]

**Pull request merged**